### PR TITLE
Customize messages when entering/exiting fractions

### PR DESCRIPTION
**Pull request by kevinbarabash**
Screen Shot 2021-05-20 at 2 51 14 PM
Screen Shot 2021-05-20 at 2 51 26 PM
Screen Shot 2021-05-20 at 3 00 23 PM
Screen Shot 2021-05-20 at 3 03 51 PM

### CODE REVIEW

File: build/mathquill-basic.js [hunk 1]
```diff
@@ -2183,29 +2187,96 @@ Controller.open(function(_) {
     prayDirection(dir);
     var cursor = this.cursor, updown = cursor.options.leftRightIntoCmdGoes;
 
+    let leaving = "";
```
Comment (alfovo) [inline build/mathquill-basic.js:2190]: Nice!!

File: src/services/keystroke.js [hunk 2]
```diff
@@ -176,29 +180,96 @@ Controller.open(function(_) {
     prayDirection(dir);
     var cursor = this.cursor, updown = cursor.options.leftRightIntoCmdGoes;
 
+    let leaving = "";
+    if (dir === R && cursor[R] === 0) {
+      const parent = cursor.parent;
+      const node = parent.jQ[0];
+
+      if (node.classList.contains("mq-denominator")) {
```
Comment (alfovo) [inline src/services/keystroke.js:188]: Lovely 👂

**Review by alfovo — APPROVED** [review]
Looks great! looks like this covers a lot of https://github.com/Khan/mathquill/issues/6

File: src/services/keystroke.js [hunk 3]
```diff
@@ -176,29 +180,96 @@ Controller.open(function(_) {
...
+        message = "you're in an empty math expression";
+      }
+    } else {
+      if (prev === 0) {
+        if (node.classList.contains("mq-numerator")) {
+          message = "you're at the start of the numerator";
+        } else if (node.classList.contains("mq-denominator")) {
+          message = "you're at the start of the denominator";
+        } else if (node) {
+          message = "you're at the start of a sub-expression";
```
Comment (alfovo) [inline src/services/keystroke.js:242]: hyper-nit: I have a slight preference for hearing full words rather than contractions in the screen reader.

Comment (kevinbarabash) [reply src/services/keystroke.js:242]: I'll change that before merging this.

**Pull request merged**